src/lsm/lsm_worker.c

```diff
@@ -93,11 +92,13 @@ __lsm_worker(void *arg)
 
 	cookie = (WT_LSM_WORKER_ARGS *)arg;
 	session = cookie->session;
-	conn = S2C(session);
 
 	entry = NULL;
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
```

Comment:
@agorrod 
> I'm personally a fan of having a pair of flags controlling thread shutdown.

I don't strongly disagree, although it potentially makes shutdown more complicated because a worker thread might exit earlier than the manager wants, for example, imagine the manager wants worker threads to finish up some work before exiting.